apps/frontend/src/modules/auth/slices/auth.slice.ts

```diff
@@ -18,16 +18,33 @@ const initialState: State = {
 
 const { actions, name, reducer } = createSlice({
 	extraReducers(builder) {
-		builder.addCase(signUp.pending, (state) => {
+		builder.addCase(getAuthenticatedUser.fulfilled, (state) => {
+			state.dataStatus = DataStatus.FULFILLED;
+		});
+		builder.addCase(getAuthenticatedUser.pending, (state) => {
 			state.dataStatus = DataStatus.PENDING;
 		});
-		builder.addCase(signUp.fulfilled, (state, action) => {
+		builder.addCase(getAuthenticatedUser.rejected, (state) => {
+			state.authenticatedUser = null;
+			state.dataStatus = DataStatus.REJECTED;
+		});
+
+		builder.addCase(signUp.fulfilled, (state) => {
 			state.dataStatus = DataStatus.FULFILLED;
-			state.authenticatedUser = action.payload;
+		});
+		builder.addCase(signUp.pending, (state) => {
+			state.dataStatus = DataStatus.PENDING;
 		});
 		builder.addCase(signUp.rejected, (state) => {
 			state.dataStatus = DataStatus.REJECTED;
 		});
+
+		builder.addMatcher(
```

Comment:
Add separate cases instead of matcher

Reply:
> Add separate cases instead of matcher

but what to do if @yevhenii-hladkov has already asked to add matches? https://github.com/BinaryStudioAcademy/bsa-2024-gitfit/pull/49#discussion_r1725067100

Reply:
oh, didn't saw it, ok then let's leave it